### PR TITLE
[9.0][FIX] Account: Allow "match payment" to work in a multicompany environment

### PR DESCRIPTION
**Pull request by ivantodorovich**
**Description of the issue/feature this PR addresses:**
It allows to use the Match Payments feature in a multicompany environment.

**Impacted versions:** 9.0, 10.0, 11.0

**Current behavior before PR:**
In a multicompany environment, te reconcile interface only works for account lines belonging to the same company the user is logged in, but it completely ignore child companies. The user is expected to log into each of the child companies if he wants to use this interface.

**Desired behavior after PR is merged:**
If the user has proper access rights, it should let him reconcile lines, even if he is not logged into the exact same company.

Somewhat related to https://github.com/odoo/odoo/pull/26782

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -489,6 +489,8 @@ def get_data_for_manual_reconciliation(self, res_type, res_ids=None, account_typ
         is_partner = res_type == 'partner'
         res_alias = is_partner and 'p' or 'a'
 
+        company_ids = self.env['res.company'].search([('id', 'child_of', self.env.user.company_id.id)])
```
[the remark this answers is not in the document]
Comment (ivantodorovich) [reply addons/account/models/account_move.py]: it's not `in` operator, it's `child_of`

[the remark this answers is not in the document]
Comment (pedrobaeza) [reply addons/account/models/account_move.py]: Sorry, my bad. I mean child_of

[the remark this answers is not in the document]
Comment (ivantodorovich) [reply addons/account/models/account_move.py]: It works either way, you can see some examples of it's use here:
https://github.com/odoo/odoo/blob/d7f5a1b25e4c14af49cccd65ad3b6e0553a77faa/addons/website_crm_partner_assign/controllers/main.py#L25
https://github.com/odoo/odoo/blob/07cdd24f30c221de96dc1dda452959f9ae358c12/addons/product/models/product.py#L49
https://github.com/odoo/odoo/blob/5b89ca142d5e932dff0dffe04d187cf5eda7ac7e/addons/purchase/models/account_invoice.py#L32

**Comment by mart-e** [comment]
Hi,

The 9.0 is no longer supported, could you retarget for upper version please?

**Comment by ivantodorovich** [comment]
@mart-e PR https://github.com/odoo/odoo/pull/28516

**Pull request closed without merging**